src/controllers/dapps/dapps.ts

```diff
@@ -46,65 +123,225 @@ export class DappsController extends EventEmitter implements IDappsController {
+
+    const [res, chainsRes] = await Promise.all([
+      this.#fetch('https://api.llama.fi/protocols'),
+      this.#fetch('https://api.llama.fi/v2/chains')
+    ])
+
+    if (!res.ok || !chainsRes.ok) {
+      throw new Error(`Fetch failed: protocols=${res.status}, chains=${chainsRes.status}`)
+    }
+
+    ;[fetchedDappsList, fetchedChainsList] = await Promise.all([res.json(), chainsRes.json()])
+
+    if (!fetchedDappsList.length || !fetchedChainsList.length) {
+      this.#shouldRetryFetchAndUpdate = true
+
+      // run the interval if the initial fetch failed while the extension is not in use
+      if (!this.#retryFetchAndUpdateAttempts && !this.#ui.views.some((v) => v.type === 'popup')) {
+        this.#retryFetchAndUpdateInterval.start()
+      } else {
+        this.#retryFetchAndUpdateInterval.stop()
+      }
+      return
+    }
+
+    const chainNamesToIds = new Map<string, number>()
+    for (const c of fetchedChainsList) {
+      chainNamesToIds.set(c.name.toLowerCase(), c.chainId)
+    }
+
+    const nonEvmChainsByName = fetchedChainsList
+      .filter((c) => !c.chainId)
+      .map((c) => c.name.toLowerCase())
+
+    const prevDapps = new Map(this.#dapps)
+
+    for (const dapp of fetchedDappsList) {
+      if (['CEX', 'Developer Tools'].includes(dapp.category)) continue
+
+      if (defiLlamaProtocolIdsToExclude.includes(dapp.id)) continue
+
+      const id = getDappIdFromUrl(dapp.url)
+
+      // Tries to find non-EVM protocols by matching their text with known non-EVM chain names because
+      // some protocols have an empty chains props, and thats the only way to filter the non-EVM ones.
+      if (categoriesNotToFilterOut.includes(dapp.category) && !dapp.chains.length) {
+        const text = `${dapp.name} ${dapp.description ?? ''}`.toLowerCase()
+        if (nonEvmChainsByName.some((chainName) => text.includes(chainName))) {
+          continue
+        }
+      }
+
+      const prevStoredDapp = prevDapps.get(id)
+
+      const chainIds = (dapp.chains ?? [])
+        .map((chainName: string) => chainNamesToIds.get(chainName.toLowerCase())!)
+        .filter(
+          (chainId: number) =>
+            !!chainId &&
+            this.#networks.allNetworks.find((n) => n.chainId.toString() === chainId.toString())
+        )
+
+      const updatedDapp: Dapp = {
+        id,
+        name: formatDappName(dapp.name),
+        description: dapp.description,
+        url: dapp.url,
+        icon: dapp.logo,
+        category: dapp.category,
+        tvl: dapp.tvl,
+        chainIds,
+        isConnected: prevStoredDapp?.isConnected || false,
+        isFeatured: featuredDapps.has(id) || featuredDapps.has(getDomainFromUrl(dapp.url)!),
+        isCustom: !!prevStoredDapp?.isCustom,
+        chainId: prevStoredDapp?.chainId || 1,
+        favorite: !!prevStoredDapp?.favorite,
+        blacklisted: !!prevStoredDapp?.blacklisted,
+        twitter: dapp.twitter,
+        geckoId: dapp.gecko_id,
```

Comment:
what do you think about moving defi lama specific data to a separate object, like:
```
extraData: { tvl, category, chainIds, twitter, geckoId }
```